ext/src/helpers/output_limited_buffer.rs

```diff
@@ -1,9 +1,63 @@
+use bytes::Bytes;
 use magnus::{
     value::{InnerValue, Opaque, ReprValue},
     RString, Ruby,
 };
-use std::io;
-use std::io::ErrorKind;
+use std::io::Write;
+use std::sync::{Arc, Mutex};
+use wasmtime_wasi::{OutputStream, Pollable, StdoutStream, StreamError, StreamResult};
+
+pub struct WritePipe {
```

Comment:
We need a custom type so we can implement `StdoutStream` which effectively requires wrapping all fields in `OutputLimitedBuffer` in the same `Arc`.